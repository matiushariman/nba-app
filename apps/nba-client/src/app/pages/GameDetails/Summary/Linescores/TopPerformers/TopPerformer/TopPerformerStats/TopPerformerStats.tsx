import type { TopPerformerStatsProps } from './TopPerformerStats.types';

const TopPerformerStats = ({ label, value }: TopPerformerStatsProps) => (
  <div>
    <p className="text-center font-bold text-xl">{value}</p>
    <p className="text-center text-sm uppercase">{label}</p>
  </div>
);

export default TopPerformerStats;
