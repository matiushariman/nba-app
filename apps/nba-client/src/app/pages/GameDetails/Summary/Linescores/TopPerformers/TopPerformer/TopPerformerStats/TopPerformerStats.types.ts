export interface TopPerformerStatsProps {
  readonly label: string;
  readonly value: string | number;
}
