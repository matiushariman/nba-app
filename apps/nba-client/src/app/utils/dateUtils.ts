import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';

dayjs.extend(utc);
dayjs.extend(timezone);

interface ConvertDateToEtParams {
  date?: string | number | dayjs.Dayjs | Date;
  format?: string;
}

export const convertDateToEt = ({
  date,
  format = 'MMMM D, YYYY h:mm A',
}: ConvertDateToEtParams) => dayjs(date).tz('America/Toronto').format(format);
