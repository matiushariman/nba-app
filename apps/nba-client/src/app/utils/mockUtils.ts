export const shouldUseMock = process.env['NODE_ENV'] === 'development';
