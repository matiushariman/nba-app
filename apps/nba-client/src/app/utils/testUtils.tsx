import { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';

import '../../styles.css';

const queryClient = new QueryClient();

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
