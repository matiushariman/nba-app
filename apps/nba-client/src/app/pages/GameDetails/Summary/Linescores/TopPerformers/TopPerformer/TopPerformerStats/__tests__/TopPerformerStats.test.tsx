import TopPerformerStats from '..';
import { render } from '../../../../../../../../utils/testUtils';

describe('pages/GameDetails/Summary/Linescores/TopPerformers/TopPerformer/TopPerformerStats', () => {
  it('should render TopPerformerStats that matches snapshot', () => {
    const { baseElement } = render(
      <TopPerformerStats label="Pts" value={100} />
    );

    expect(baseElement).toMatchSnapshot();
  });
});
