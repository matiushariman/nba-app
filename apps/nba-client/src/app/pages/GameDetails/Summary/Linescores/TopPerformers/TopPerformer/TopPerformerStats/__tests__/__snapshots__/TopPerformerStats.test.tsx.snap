// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/GameDetails/Summary/Linescores/TopPerformers/TopPerformer/TopPerformerStats should render TopPerformerStats that matches snapshot 1`] = `
<body>
  <div>
    <div>
      <p
        class="text-center font-bold text-xl"
      >
        100
      </p>
      <p
        class="text-center text-sm uppercase"
      >
        Pts
      </p>
    </div>
  </div>
</body>
`;
